app/models/tx-store/tx-store.ts

```diff
@@ -170,25 +171,25 @@ export const TxStoreModel = types
           self.isCreatingTx = false
         }
       }),
-      signTx: flow(function * (signer: any) {
+      signTx: flow(function* (signer: OfflineDirectSigner) {
         if (self.isSigningTx) return
 
         self.isSigningTx = true
         self.errorMessage = ""
         self.txHash = ""
         try {
-          const { hash, included }: CosmosSendResult = yield message.send(self.getMeta(), signer)
-          self.txHash = hash
+          const client: CosmosSigner = yield self.env.cosmosAPI.createSigningClient(signer)
+          const response: BroadcastTxResponse = yield client.signAndBroadcast({ ...message, ...self.getMeta() })
+          self.txHash = response.transactionHash
           // TODO: Store hash for history
-          const response: CosmosTxQueryResult = yield included()
-          if (response?.logs[0]?.success) {
+          if (!('code' in response) || !response.code) {
```

Comment:
Can be simplify this condition statement?

Reply:
How about I keep only `if (!response.code)`? cc @williamchong

Reply:
ok if typescript doesn't complain about non existent field "code'

Reply:
Seems it will, so I would replace with `if (!('code' in response))`.